compiler/rustc_resolve/src/imports.rs

```diff
@@ -308,7 +309,20 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                 }
                 match (old_binding.is_glob_import(), binding.is_glob_import()) {
                     (true, true) => {
-                        if res != old_binding.res() {
+                        if binding.is_ambiguity() {
+                            resolution.binding =
+                                Some(self.arenas.alloc_name_binding(NameBindingData {
+                                    warn_ambiguity: true,
```

Comment:
maybe not. 

For example, in ﻿`issue-47525.rs`, there is a scenario where an error should be thrown but is not in the master. And it will generate a warning after this PR. The corresponding code is as follows:

```rs
fn main() {
    use a::*;
    x();    // should warning here
}

mod a {
    mod b {
        pub fn x() {  }
    }
    mod c {
        pub fn x() { }
    }

    pub use self::b::*;
    pub use self::c::*;
}
```

The order of ﻿`update_resolution` for the `﻿x` in the diff module is as follows:

| order| `module` | `old_binding` | `binding` |  `binding` after update |
| - | - | - | - | - |
| 1 | `root::a::b` | None |   `root::a::b::x` | same as left |
| 2 | `root::a::c` | None |   `root::a::c::x` | same as left |
| 3 | `root::a`     | None |  `root::a::b::x`  | same as left |
| 4 | `root`         | None |  `root::a::b::x`  | same as left |  
| 5 | `root::a`     | `root::a:b::x` |  `root::a::c::x`  |`root::a:b::x` with ambiguity error |
| 6| `root`          | `root::a::b::x` without error |  `root::a:b::x` with ambiguity error |  ? | 

Focus on this last line:

- if the first condition is `if binding.is_ambiguity() { xxx }` is will generator a warning for ambiguity.
- However, if the first condition is ` xxx && old_import == import { xx }`, it will update `Mod(root), Ident(x)` to the `root::a:b::x` with ambiguity error, resulting in an eventual error being thrown.

Reply:
>it will update `Mod(root), Ident(x)` to the `root::a:b::x` with ambiguity error, resulting in an eventual error being thrown.

Why will the error be thrown?
`warn_ambiguity: true` should turn that error into a warning.

Reply:
For this case, it will never allocate a binding with a warning for ambiguity.
Nevertheless, I have updated this code, and it now appears more reasonable.